src/core/thread/mle.cpp

```diff
@@ -641,6 +641,13 @@ void Mle::SetStateDetached(void)
     mDataRequestAttempts     = 0;
     Get<MeshForwarder>().SetRxOnWhenIdle(true);
     Get<Mac::Mac>().SetBeaconEnabled(false);
+#if OPENTHREAD_CONFIG_MAC_CSL_RECEIVER_ENABLE
+    if (Get<Mac::Mac>().IsCslEnabled())
+    {
+        // Turn off CSL in radio driver when child becomes detached.
```

Comment:
Comment may not be needed here since it's obvious.

Reply:
Fixed.